### PR TITLE
Overfitting/Underfitting tutorial

### LINKED ISSUE #197: Missing tutorial: underfitting and overfitting (learning curve)
We should cover the use case with ske so it's easier for users to tackle the issue when it occurs.

### PR DESCRIPTION
**Pull request by gtduncan**
## Describe your changes
Tutorial on underfitting/overfitting implementing learning curves created by sk-eval

## Issue ticket number and link
Closes #197 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added thorough [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [x] I have added the right documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)

----
:books: Documentation preview :books:: https://sklearn-evaluation--299.org.readthedocs.build/en/299/

### CODE REVIEW

File: docs/_toc.yml [hunk 1]
```diff
@@ -60,4 +60,8 @@ parts:
     - file: reference/community
     - file: reference/developer
     - file: reference/developer-api
+  
+  - caption: Tutorials
+    chapters:
+    - file: tutorials/underfitting_overfitting
```
Comment (edublancas) [inline docs/_toc.yml:66]: let's put this under the optimization section

File: docs/tutorials/underfitting_overfitting.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+If there is a large gap between the train and validation curve, we are overfitting.
+
+Here is an example of a learning curve with significant overfitting:
+
+```{code-cell} ipython3
+data = make_classification(
+    n_samples=1000, random_state=0, class_sep=0.2
+)
+X, y = data[0], data[1]
```
Comment (edublancas) [inline docs/tutorials/underfitting_overfitting.md:77]: cleaner version

```suggestion
X, y = make_classification(
    n_samples=1000, random_state=0, class_sep=0.2
)
```

File: docs/tutorials/underfitting_overfitting.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+If we see that the training score is consistently and particularly low, we are underfitting.
+
+Here is an example of a learning curve with underfitting:
+
+```{code-cell} ipython3
+data = make_classification(
+    n_samples=100, n_features=20, n_informative=2,
+    n_repeated=2, n_redundant=2, class_sep=0, random_state=0
+)
+X, y = data[0], data[1]
```
Comment (edublancas) [inline docs/tutorials/underfitting_overfitting.md:105]:
```suggestion
X, y = make_classification(
    n_samples=100, n_features=20, n_informative=2,
    n_repeated=2, n_redundant=2, class_sep=0, random_state=0
)
```

File: docs/tutorials/underfitting_overfitting.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+There are a few ways in which we can improve our performance. As some solutions can help with both and some are relevant only to one set of issues, we'll seperate the methods into 2 sections.
+
+### Solving Overfitting
+
+1) Increase your dataset's size through augmentation. Applying transformations to or reworking existing data can greatly improve your model's performance.
+2) [Regularization techniques](https://cmci.colorado.edu/classes/INFO-4604/files/slides-6_regularization.pdf) such as dropout and early stopping can be very useful.
+
+Some other methods may include:
+
+cross-validation, ensemble methods
```
Comment (edublancas) [inline docs/tutorials/underfitting_overfitting.md:138]:
```suggestion
Some other methods include: cross-validation, and ensemble methods.
```

File: docs/tutorials/underfitting_overfitting.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+### Solving Underfitting
+
+1) Add more observations. More data may enable your model to pick up on existing patterns.
+2) Add more features. Your dataset may be too simple, and your model may not have enough features to create an optimal fit.
+3) Select a more powerful and/or complex model.
+
+Some other methods may include:
+
+ensemble methods, [hyperparameter tuning](https://guides.lib.purdue.edu/d-velop/ml-dl/automl1), transfer learning
```
Comment (edublancas) [inline docs/tutorials/underfitting_overfitting.md:148]:
```suggestion
Some other methods include: ensemble methods, [hyperparameter tuning](https://guides.lib.purdue.edu/d-velop/ml-dl/automl1), and transfer learning.
```

**Review by edublancas — CHANGES_REQUESTED** [review]
looks great, added minor observations

**Comment by gtduncan** [comment]
Just made those changes!

**Comment by idomic** [comment]
@gtduncan please resolve the `CHANGELOG.md` conflicts so we can merge.

**Review by edublancas — APPROVED** [review]

**Pull request merged**